File: 2024/day1.js
const fs = require('node:fs');


try {
    const data = fs.readFileSync('inputs/day1.txt', 'utf8');

    let lines = data.split("\r\n")
    let totalDistance = 0;
    let totalLines = lines.length;
    let array1 = [];
    let array2 = [];
    let array3 = [];
    let similarityScore = 0;
    lines.forEach(function(line,index){
        let parts = line.split("   ")
        array1.push(parts[0]);
        array2.push(parts[1]);
    })
    array1.sort((a,b) => a - b); //sorts smallest to biggest
    array2.sort((a,b) => a - b);
    array1.forEach(function(number,index){
       let distance = Math.abs( array1[index] - array2[index]);
        array3.push( distance );
        totalDistance+=distance;
        let valueToLookFor = array1[index]
        let occurrences = array2.filter((v) => (v === valueToLookFor)).length
        let similarity = valueToLookFor * occurrences;
        similarityScore +=similarity;
    })
    console.log(array1,array2,array3);
    console.log("Part 1: " + totalDistance);
    console.log("Part 2: " + similarityScore);

} catch (err) {
    console.error(err);
}
// Solved
// Part 1: 3714264
// Part 2: 18805872

File: 2022/day1.go
package main

import (
	"fmt"
	"strings"
	"os"
	"strconv"
	"sort"
)
func check(e error) {
    if e != nil {
		fmt.Println(e)
        panic(e)
    }
}
func main(){
	data, err := os.ReadFile("inputs/day1.txt")
	check(err)
	str := string(data)
	groups := strings.Split(str, "\r\n\r\n")
	maxCals := 0
	elves := make([]int,0)
	for _, group := range groups {
		lines := strings.Split(group, "\r\n")
		groupCals := 0
		for _, line := range lines {
			cals,_ := strconv.Atoi(line)
			groupCals += cals
		}
		if maxCals < groupCals {
			maxCals = groupCals
		}
		elves = append(elves,groupCals)
	}
	sort.Ints(elves);
	part2Total := 0
	lastThree := elves[(len(elves) -3 ):]
	for _, number := range lastThree {
		part2Total+=number
	}

	fmt.Println(maxCals)
	fmt.Println(part2Total)

}
//part 1: 70720
//p2: 207148